File: Question-12.cpp
#include <iostream>
using namespace std;

class Employee
{
public:
    string name;
    int age;
    double salary;
    Employee *next;
};

class EmployeeList
{
    Employee *head;
    Employee *tail;

    int size;

public:
    EmployeeList()
    {
        head = NULL;
        tail = NULL;
        size = 0;
    }

    void addEmployee()
    {
        string name;
        cout << "Enter the name : ";
        cin >> name;

        int age;
        cout << "Enter the age : ";
        cin >> age;

        double salary;
        cout << "Enter the salary : ";
        cin >> salary;

        Employee *newEmployee = new Employee();

        newEmployee->name = name;
        newEmployee->age = age;
        newEmployee->salary = salary;
        newEmployee->next = NULL;

        if (head == NULL)
        {
            head = newEmployee;
            tail = newEmployee;
        }

        else
        {
            tail->next = newEmployee;
            tail = newEmployee;
        }

        size++;
    }
    
    void printEmployeeList()
    {
        Employee *temp = head;
        while (temp != NULL)
        {
            cout << "Employee Name : " << temp->name << "\n";
            cout << "Employee Age : " << temp->age << "\n";
            cout << "Employee Salary : " << temp->salary << "\n";
            temp = temp->next;
        }
    }

    void searchEmployee()
    {
        string name;
        cout << "Enter the name : ";
        cin >> name;

        Employee *temp = head;

        while (temp != NULL)
        {
            if (temp->name == name)
            {
                cout << temp->name << " " << temp->age << " " << temp->salary << "\n";
                return;
            }

            temp = temp->next;
        }

        cout << "Employee not found!!!\n";
    }
};

int main()
{
    EmployeeList *e = new EmployeeList;

    e->addEmployee();
    
    int choice;

    while (true)
    {
        cout << "Do you want to continue? (Press '1' to continue, '0' to break) : ";
        cin >> choice;

        if (choice == 0)
            break;

        else
            e->addEmployee();
    }

    cout << "\nEmployee List :-\n";
    e->printEmployeeList();
    cout << "\n";

    e->searchEmployee();
    e->searchEmployee();

    return 0;
}

File: Question-12.java
import java.util.Scanner;

class Employee {

    public Scanner sc = new Scanner(System.in);

    public String name;
    public int age;
    public double salary;

    public Employee next;
    public Employee link;

    Employee() {

    }

    Employee(String name, int age, double salary) {

        this.name = name;
        this.age = age;
        this.salary = salary;
    }

    void details() {
        System.out.println("Name : " + name);
        System.out.println("Age : " + age);
        System.out.println("Salary : " + salary);
    }
}

class Employee_List {

    public Scanner sc = new Scanner(System.in);

    Employee start;

    Employee_List() {
        start = new Employee();
    }

    void setData() {

        String s;
        int a;
        double d;
        int choice;

        Employee ptr = start;

        do {
            System.out.print("Name : ");
            s = sc.nextLine();

            System.out.print("Age : ");
            a = sc.nextInt();

            System.out.print("Salary : ");
            d = sc.nextDouble();

            ptr.next = new Employee(s, a, d);
            ptr = ptr.next;

            System.out.print("Press '1' to continue : ");
            choice = sc.nextInt();
        } while (choice != 0);
    }

    void display() {
        Employee ptr = start.link;

        while (ptr != null) {
            ptr.details();
            ptr = ptr.link;
        }
    }
}

class List_Employee {

    public static void main(String args[]) {

        Employee_List el = new Employee_List();
        el.setData();
        el.display();
    }
}